mirascope/base/ops_utils.py

```diff
@@ -0,0 +1,259 @@
+import inspect
+from collections.abc import Callable
+from contextlib import AbstractContextManager
+from functools import wraps
+from typing import Any, Optional, TypeVar, Union
+
+from pydantic import BaseModel
+
+from mirascope.rag.embedders import BaseEmbedder
+
+from .calls import BaseCall
+
+
+def get_class_vars(cls: BaseModel) -> dict[str, Any]:
+    """Get the class variables of a `BaseModel` removing any dangerous variables."""
+    class_vars = {}
+    for classvars in cls.__class_vars__:
+        if not classvars == "api_key":
+            class_vars[classvars] = getattr(cls.__class__, classvars)
+    return class_vars
+
+
+T = TypeVar("T")
+
+
+def get_wrapped_async_client(client: T, self: Union[BaseCall, BaseEmbedder]) -> T:
+    """Get a wrapped async client."""
+    if self.configuration.client_wrappers:
+        for op in self.configuration.client_wrappers:
+            if op == "langfuse":  # pragma: no cover
+                from langfuse.openai import AsyncOpenAI as LangfuseAsyncOpenAI
+
+                client = LangfuseAsyncOpenAI(
+                    api_key=self.api_key, base_url=self.base_url
+                )
+            elif op == "logfire":  # pragma: no cover
+                import logfire
+
+                logfire.instrument_openai(client)  # type: ignore
+            elif callable(op):
+                client = op(client)
+    return client
+
+
+def get_wrapped_client(client: T, self: Union[BaseCall, BaseEmbedder]) -> T:
+    """Get a wrapped client."""
+    if self.configuration.client_wrappers:
+        for op in self.configuration.client_wrappers:  # pragma: no cover
+            if op == "langfuse":
+                from langfuse.openai import OpenAI as LangfuseOpenAI
+
+                client = LangfuseOpenAI(api_key=self.api_key, base_url=self.base_url)
+            elif op == "logfire":  # pragma: no cover
+                import logfire
+
+                logfire.instrument_openai(client)  # type: ignore
+            elif callable(op):
+                client = op(client)
+    return client
+
+
+C = TypeVar("C")
+
+
+def get_wrapped_call(call: C, self: Union[BaseCall, BaseEmbedder], **kwargs) -> C:
+    """Wrap a call to add the `llm_ops` parameter if it exists."""
+
+    if self.configuration.llm_ops:
```

Comment:
aha, since we call this here, we can remove all other checks in all the calls right?